app/src/main-process/ipc-main.ts

```diff
@@ -40,10 +45,22 @@ export function once<T extends keyof RequestChannels>(
  */
 export function handle<T extends keyof RequestResponseChannels>(
   channel: T,
-  listener: (
-    event: IpcMainInvokeEvent,
-    ...args: Parameters<RequestResponseChannels[T]>
-  ) => ReturnType<RequestResponseChannels[T]>
+  listener: RequestResponseChannelListener<T>
 ) {
-  ipcMain.handle(channel, (event, ...args) => listener(event, ...(args as any)))
+  ipcMain.handle(channel, safeListener(listener))
+}
+
+const validSenderUrl = encodePathAsUrl(__dirname, 'index.html').toString()
+
+function safeListener(listener: (...args: any[]) => any) {
+  return (event: IpcMainEvent | IpcMainInvokeEvent, ...args: any[]) => {
+    if (event.senderFrame.url !== validSenderUrl) {
+      log.error(
+        `IPC message received from invalid sender: ${event.senderFrame.url}`
+      )
+      return
+    }
+
+    return listener(event, ...args)
+  }
```

Comment:
You could do something like this to at least enforce that the listener takes an IPC event as its first parameter. Not much type safety but a tad bit more at least

```suggestion
function safeListener<E extends IpcMainEvent | IpcMainInvokeEvent, R>(
  listener: (event: E, ...a: any) => R
) {
  return (event: E, ...args: any) => {
    if (event.senderFrame.url !== validSenderUrl) {
      log.error(
        `IPC message received from invalid sender: ${event.senderFrame.url}`
      )
      return
    }
    return listener(event, ...args)
  }
```

Reply:
Love it, thanks! ❤️